habitat/tasks/nav/multi_object_nav_task.py

```diff
@@ -0,0 +1,551 @@
+        observations.update(
+            self.sensor_suite.get_observations(
+                observations=observations,
+                episode=episode,
+                action=action,
+                task=self,
+            )
+        )
+
+        self._is_episode_active = self._check_episode_is_active(
+            episode=episode
+        )
+
+        return observations
+
+    def _check_episode_is_active(
+        self, episode: MultiObjectGoalNavEpisode
+    ) -> bool:
+        current_position = self._sim.get_agent_state().position.tolist()
+        distance_to_current_goal = self._sim.geodesic_distance(
+            current_position,
+            episode.goals[self.current_goal_index].position,
+        )
+        is_episode_active = not (
+            self.is_found_called
+            and distance_to_current_goal > self._config.SUCCESS_DISTANCE
+        )
+        return is_episode_active
+
+
+@registry.register_sensor
+class MultiObjectGoalSensor(Sensor):
+    r"""A sensor for Multi Object Goal specification as observations which is used in
+    MultiObjectGoalNavigation. The goal is expected to be specified by object_id.
+    Args:
+        sim: a reference to the simulator for calculating task observations.
+        config: a config for the ObjectGoalSensor sensor. Can contain field
+            GOAL_SPEC_MAX_VAL the maximum object_id possible used for
+            observation space definition.
+        dataset: a Multi Object Goal navigation dataset that contains dictionaries
+        of categories id to text mapping.
+    """
+    cls_uuid: str = "multiobjectgoal"
+
+    def __init__(
+        self,
+        sim,
+        config: Config,
+        dataset: "MultiObjectNavDatasetV1",
+        *args: Any,
+        **kwargs: Any,
+    ):
+        self._dataset = dataset
+        super().__init__(config=config)
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "multiobjectgoal"
+
+    def _get_sensor_type(self, *args: Any, **kwargs: Any):
+        return SensorTypes.SEMANTIC
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        sensor_shape = (1,)
+        max_value = max(self._dataset.category_to_task_category_id.values())
+
+        return spaces.Box(
+            low=0, high=max_value, shape=sensor_shape, dtype=np.int64
+        )
+
+    def get_observation(
+        self,
+        observations,
+        *args: Any,
+        episode: MultiObjectGoalNavEpisode,
+        task: MultiObjectNavigationTask,
+        **kwargs: Any,
+    ) -> Optional[int]:
+        category_name = [i.object_category for i in episode.goals]
+        goals = np.array(
+            [
+                self._dataset.category_to_task_category_id[i]
+                for i in category_name
+            ],
+            dtype=np.int64,
+        )
+        return goals[task.current_goal_index : task.current_goal_index + 1]
+
+
+@registry.register_measure
+class DistanceToCurrentObjectGoal(Measure):
```

Comment:
I think this can be rolled into the existing DistanceToGoal measure.  That measure is ill-defined for MultiON, so defining it to be the current goal makes sense.